### PR TITLE
airbyte-ci: use connectors-qa instead of connector_ops.qa_check

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte-internal-issues/issues/2905

This PR replaces the use of `connector_ops.qa_checks.py` for the new `connectors_qa` package (https://github.com/airbytehq/airbyte/pull/35322) in the `QAChecks` step which run on `airbyte-ci connectors test`

## How
* Call `connectors_qa` via a `SimpleDockerStep`
* Remove redundant step which are already checks in `connectors_qa`
Screenshot 2024-02-15 at 16 28 41

## Manual testing
* Manually run `airbyte-ci connectors --name=source-faker test` to make sure no regression is introduced by the deletion of some steps. [Ran](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/manual/augustin_02-12-airbyte-ci_use_connectors-qa_instead_of_connector_opsqa_check/1708095830/aa30a14e5ed3216fd793d38a545bc488fb0beaed/source-amazon-ads/4.0.3/output.html) on `source-amazon-ads` which is 🟢 on `master`
* Manually run a pre-release of a connector for extra safety

### CODE REVIEW

**Comment by alafanechere** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35325?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* `master`
  * **#35322** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35322?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **#35325** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35325?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **#35324** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py [hunk 1]
```diff
@@ -345,31 +310,3 @@ async def _run(self, *args: Any, **kwargs: Any) -> StepResult:
                 stdout=f"Connector is certified but is still using a Dockerfile. {migration_hint}",
             )
         return StepResult(step=self, status=StepStatus.SUCCESS, stdout="Connector is certified and uses our base image.")
-
-
-class CheckPythonRegistryPublishConfiguration(Step):
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py:350]: This is now checked inside the `connectors_qa` package https://github.com/airbytehq/airbyte/pull/35322

File: airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py [hunk 2]
```diff
@@ -119,71 +118,37 @@ def validate(self) -> StepResult:
         return self.success_result
 
 
-class VersionFollowsSemverCheck(VersionCheck):
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py:122]: This is now checked in the `connectors_qa` package https://github.com/airbytehq/airbyte/pull/35322

File: airbyte-ci/connectors/pipelines/tests/test_tests/test_common.py [hunk 3]
```diff
@@ -243,116 +243,3 @@ async def test_params(self, dagger_client, mocker, test_context_ci, test_input_d
         assert set(acceptance_test_step.params_as_cli_options) == {"-ra", "--disable-warnings", "--durations=3"}
         acceptance_test_step.extra_params = {"--durations": ["5"], "--collect-only": []}
         assert set(acceptance_test_step.params_as_cli_options) == {"-ra", "--disable-warnings", "--durations=5", "--collect-only"}
-
-
-class TestCheckBaseImageIsUsed:
```
Comment (alafanechere) [inline airbyte-ci/connectors/pipelines/tests/test_tests/test_common.py:248]: This is now checked in the `connectors_qa` package https://github.com/airbytehq/airbyte/pull/35322

Comment (maxi297) [inline airbyte-ci/connectors/pipelines/tests/test_tests/test_common.py:248]: I don't see any usage of `CheckBaseImageIsUsed` and we are removing the tests. Should we remove [the implementation](https://github.com/airbytehq/airbyte/blob/414b9aded25fd14e0253c97242a3c2f68f801f79/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py#L288-L312) as well?

File: airbyte-ci/connectors/pipelines/pipelines/dagger/containers/internal_tools.py [hunk 4]
```diff
@@ -37,3 +37,17 @@ async def with_connector_ops(context: PipelineContext) -> Container:
     python_base_environment: Container = with_python_base(context)
 
     return await with_installed_pipx_package(context, python_base_environment, INTERNAL_TOOL_PATHS.CONNECTOR_OPS.value)
+
+
+async def with_connectors_qa(context: PipelineContext) -> Container:
```
Comment (maxi297) [inline airbyte-ci/connectors/pipelines/pipelines/dagger/containers/internal_tools.py:42]: For my personal knowledge: how is this used?

**Review by maxi297 — COMMENTED** [review]
From my limited knowledge, I'm fine with this. 

One concern: `test_cat_container_caching` is currently failing in the CI. Is this a flaky test or a consequence of those changes?

**Comment by alafanechere** [comment]
> One concern: test_cat_container_caching is currently failing in the CI. Is this a flaky test or a consequence of those changes?

Yes it's flaky, I believe I observe flakyness when multiple branches triggers the test suite in CI.

File: airbyte-ci/connectors/pipelines/tests/test_tests/test_common.py (hunk 3 above)
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/tests/test_tests/test_common.py:248]: Oh yes good catch, I'll remove the implementation.

File: airbyte-ci/connectors/pipelines/pipelines/dagger/containers/internal_tools.py (hunk 4 above)
Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/dagger/containers/internal_tools.py:42]: It's not used, I should remove it. It's not necessary as it's the `SimpleDockerStep` which will take care of installing the `connectors_qa` package.

**Review by maxi297 — APPROVED** [review]
Thanks for the responses on the review! 🚢 

> Yes it's flaky, I believe I observe flakyness when multiple branches triggers the test suite in CI.

I think this is quite annoying to debug so I wouldn't do it now but do you think it is worth tracking how flaky this is?

If so, I would suggest two action items:
* Create an issue
* Add information in the test (maybe the assert text) to indicate to the dev that it is known as flaky and if this fails for invalid reasons, flag it in the GitHub ticket

That way, we can track how much of a pain this is and take action in due time.

**Pull request merged**